### PR TITLE
Fix processors

### PR DESCRIPTION
**Pull request by vaibhavchellani**

### CODE REVIEW

File: core/pubkey.go [hunk 1]
```diff
@@ -10,10 +10,11 @@ type Pubkey []byte
 const pubkeyLength = 128
 
 func NewPubkey(p [4]*big.Int) Pubkey {
```
Comment (vaibhavchellani) [inline core/pubkey.go:12]: @kilic can I request a quick review of this function?

File: bazooka/bazooka.go [hunk 2]
```diff
@@ -45,14 +46,11 @@ const (
 	WITHDRAW_ROOTS = "withdrawRoots"
 )
 
-// Global bazooka Object
```
Comment (vaibhavchellani) [inline bazooka/bazooka.go]: Removed usage of globals.

File: db/db.go [hunk 3]
```diff
@@ -35,13 +37,11 @@ type IDB interface {
 	GetAccountCollection() *mgo.Collection
 }
 
-// global DB instance created while doing init
```
Comment (vaibhavchellani) [inline db/db.go:38]: removed usage of globals

File: bazooka/bazooka.go [hunk 4]
```diff
@@ -0,0 +1,351 @@
...
+		RollupContract  *rollup.Rollup
+		State           *state.State
+		Transfer        *transfer.Transfer
+		Create2Transfer *create2transfer.Create2transfer
+		MassMigration   *massmigration.Massmigration
+		AccountRegistry *accountregistry.Accountregistry
+	}
+)
+
+// NewPreLoadedBazooka creates
```
Comment (ChihChengLiang) [inline bazooka/bazooka.go:70]: nitpick: looks like you were about to document something but didn't finish for some reason.

File: core/params.go [hunk 5]
```diff
@@ -2,13 +2,15 @@ package core
...
+	"github.com/jinzhu/gorm"
+	uuid "github.com/satori/go.uuid"
 )
 
 // Params stores all the parameters which are maintained on-chain and keeps updating
 // them whenever they change on-chain
 type Params struct {
-	DBModel
-
+	ID string `json:"-" gorm:"primary_key;size:100;default:'6ba7b810-9dad-11d1-80b4-00c04fd430c8'"`
```
Comment (ChihChengLiang) [inline core/params.go:13]: Can you elaborate what is `6ba7b810-9dad-11d1-80b4-00c04fd430c8`?

File: core/pubkey.go (hunk 1 above)
Comment (ChihChengLiang) [reply core/pubkey.go:12]: Looks about right

File: go.mod [hunk 6]
```diff
@@ -50,6 +50,7 @@ require (
 	github.com/t-tiger/gorm-bulk-insert v1.3.0
 	github.com/tdakkota/asciicheck v0.0.0-20200416200610-e657995f937b // indirect
 	github.com/tendermint/tendermint v0.32.7
+	github.com/tendermint/tm-db v0.2.0
```
Comment (ChihChengLiang) [inline go.mod:53]: Do we use this dependency anywhere?

File: bazooka/merkleProof.go [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+
+import "github.com/BOPR/core"
+
+type StateMerkleProof struct {
+	State    core.UserState
+	Siblings []core.UserState
+}
+
+func NewStateMerkleProof(account core.UserState, siblings []core.UserState) StateMerkleProof {
+	return StateMerkleProof{State: account, Siblings: siblings}
```
Comment (ChihChengLiang) [inline bazooka/merkleProof.go:11]:
```suggestion
func NewStateMerkleProof(state core.UserState, siblings []core.UserState) StateMerkleProof {
	return StateMerkleProof{State: state, Siblings: siblings}
```

**Review by ChihChengLiang — APPROVED** [review]
Awesome work!
I only did a quick look and gave some nitpicks due to the size of the PR.
The codebase looks more organized! 
Looks good to me :shipit:

File: core/params.go (hunk 5 above)
Comment (vaibhavchellani) [reply core/params.go:13]: Yeah, it's the namespace for `DNS`the lib set this up.

ctrl+f on https://tools.ietf.org/html/rfc4122.html#section-4.3

File: go.mod (hunk 6 above)
Comment (vaibhavchellani) [reply go.mod:53]: Good spot

**Comment by vaibhavchellani** [comment]
Thanks for the review! Apologies for the large PR.

**Pull request merged**